src/mongo/db/repl/replication_coordinator_impl.cpp

```diff
@@ -4172,6 +4198,29 @@ void ReplicationCoordinatorImpl::_updateWriteAbilityFromTopologyCoordinator(
     _readWriteAbility->setCanAcceptNonLocalWrites(lk, opCtx, canAcceptWrites);
 }
 
+std::vector<ReplicationCoordinator::NodeInfo> ReplicationCoordinatorImpl::getNodeInfoList() {
```

Comment:
Neither the `ReplicationCoordinator` nor the `TopologyCoordinator` have a data structure which tracks the state + configuration of every node in the replica set, which is what the `TenantSplitDonorService` needs to wait for 3 non-voting (config) nodes to finish initial sync (state). While this solution is somewhat inefficient in that a new list of `NodeInfo` is created on each call to this method, I tried to make the API opaque to a future improvement which returns cached information.